### ISSUE TITLE
coffeescript classes and es6/babel classes aren't compatible

### ISSUE THREAD

**Issue by juliankrispel** [juliankrispel/decaf#127]
cc @eventualbuddha 

Right. So at rainforest we have a legacy coffeescript/backbone codebase (which I'm trying to rid ourselves of). I've observed two problems that are a major PITA, in regards to classes.

## Problem 1: Es6 inheritance doesn't include static members but CS inheritance does

Given the example

```coffeescript
class A

A.b = [1,2,3]

class B extends A

console.log(B.b) // yields [1,2,3]
```
The above works in coffeescript because of the `extend` helper function in coffeescript. It copies the prototype as well as any `static` members of the class.

However in es6, class inheritance doesn't include static members. This means only the prototype is copied.

## Problem 2: In Coffeescript, static class properties are mounted before the constructor is called, in es6 it's the other way around.

Given the example
```js
function A(){
  console.log(this.routes); // yields undefined
}

class B extends A {
  routes = ['dwq', 'dwq']
}

new B({routes: [1,2,3]});
```
the static attribute routes will actually not be mounted when constructor `A()` is executed.

@eventualbuddha any ideas how to solve these two? Or maybe you've already addressed these?

**Comment by eventualbuddha** [comment]
For problem 1, I don't think that's actually an issue. [This example](https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-1%2Cstage-2&code=class%20A%20%7B%0A%20%20static%20b%20%3D%20%5B1%2C%202%2C%203%5D%0A%7D%0A%0Aclass%20B%20extends%20A%20%7B%7D%0A%0Aconsole.log(B.b)%3B) works the same way as the CoffeeScript one:

```js
class A {
  static b = [1, 2, 3]
}

class B extends A {}

console.log(B.b);
```

For problem 2, the `routes = …` part becomes something like this in a constructor: `this.routes = …`, but that has to happen after the implicit `super()` call. So it works in CoffeeScript because the `routes = …` becomes part of the prototype rather than a constructor assignment. I haven't done anything to address this 😞 I tend to favor the idiomatic approach over the exactly compatible one. ¯\_(ツ)_/¯

**Comment by juliankrispel** [comment]
@eventualbuddha number 2 is another candidate for an error I think

**Comment by lemonmade** [comment]
2 should probably be an error, assigning to the prototype does not have the same semantics as assigning to each instance (in the CoffeeScript case, the value is shared between all instances of the class, whereas instance class properties are set individually for each instance).

**Comment by juliankrispel** [comment]
also fixed by

**Comment by juliankrispel** [comment]
fixed in #144
